phoenix-core/src/main/java/org/apache/phoenix/calcite/CalciteUtils.java

```diff
@@ -1120,8 +1120,11 @@ public static SQLException unwrapSqlException(SQLException root){
         Exception e = root;
         while(e.getCause() != null){
             e = (Exception) e.getCause();
+            if(e instanceof RuntimeException && e.getCause() instanceof SQLException) {
```

Comment:
Latest changes attempt to locate the nested SQLException within an RuntimeException. We still throw the first SQLException seen in case there's no wrapping argument exception. And if no matches are found, we throw the root so as to avoid unexpected behavior.